nixos/modules/services/audio/ympd.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ympd;
+in {
+
+  ###### interface
+
+  options = {
+
+    services.ympd = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable ympd, the MPD Web GUI.";
+      };
+
+      webPort = mkOption {
+        default = 8080;
+        description = "The port where ympd's web interface will be available.";
+      };
+
+      user = mkOption {
+        default = "ympd";
+        description = "User account under which ympd runs.";
+      };
+
+      group = mkOption {
+        default = "ympd";
+        description = "Group to assign to ympd user.";
+      };
+
+      mpd = {
+        host = mkOption {
+          default = "localhost";
+          description = "The host where MPD is listening.";
+        };
+
+        port = mkOption {
+          default = 6600;
+          description = "The port where MPD is listening.";
+        };
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.services.ympd = {
+      after = [ "sound.target" "mpd.service" ];
+      description = "Standalone MPD Web GUI written in C";
+      wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.ympd ];
+      script = "exec ympd --host ${cfg.mpd.host} --port ${toString cfg.mpd.port} --webport ${toString cfg.webPort} --user ${cfg.user}";
+    };
+
+    users.extraUsers = optionalAttrs (cfg.user == "ympd") (singleton {
+      name = "ympd";
+      group = cfg.group;
+      extraGroups = [ "audio" ];
```

Comment:
Is the audio group really needed? Only processes that need to access the sound devices need this, but in this case only mpd would access those, right?

Also, does ympd access or save any files that need special permissions or are private? If not, a new user and group probably aren't needed and you can just run the service as nobody:nogroup.